fre/app/generate_time_averages/cdoTimeAverager.py

```diff
@@ -9,7 +9,17 @@ class inheriting from abstract base class timeAverager
     '''
 
     def generate_timavg(self, infile=None, outfile=None):
-        ''' use cdo package routines via python bindings '''
+        """
+        use cdo package routines via python bindings        
+
+        :param self: 
```

Comment:
I think the recommendation will be to document this as `:param self: This is an instance of the class cdoTimeAverager` and I would not define a type for it.

So you can keep self as the first parameter and document it here, and then delete lines 20 and 21 I think.